doc/config_reload/config_reload_enhancement.md

```diff
@@ -0,0 +1,121 @@
+Feb 8 21:13:48.654405 sonic ERR syncd#SDK: :- logEventData: fv: SAI_SWITCH_ATTR_FDB_EVENT_NOTIFY: 0x55dabccda690
+Feb 8 21:13:48.654971 sonic ERR syncd#SDK: :- logEventData: fv: SAI_SWITCH_ATTR_PORT_STATE_CHANGE_NOTIFY: 0x55dabccda6a0
+Feb 8 21:13:48.655165 sonic ERR syncd#SDK: :- logEventData: fv: SAI_SWITCH_ATTR_BFD_SESSION_STATE_CHANGE_NOTIFY: 0x55dabccda6b0
+Feb 8 21:13:48.655338 sonic ERR syncd#SDK: :- logEventData: fv: SAI_SWITCH_ATTR_SWITCH_SHUTDOWN_REQUEST_NOTIFY: 0x55dabccda6c0
+```
+This problem is solved during the reboot by selectively delaying non critical services like snmp, telemetry, etc,. using systemd timers with fixed delay. This approach is not event driven and may not be optimal during config reload.
+In this new design, the non critical services will be started only after the ports are initialized using an event driven approach.
+
+### Requirements
+
+Primary requirements for sequencing the config reload are
+- Immediately restart the critical services during config reload.
+- The non critical should be started only after all the ports are initialized.
+- Services can be configured to be started immediately or delayed. This can be using a field in FEATURE table.
+- The existing timers should be removed by this event driven approach.
+- This flow is applicable in case of all reboots (warm/fast/cold) as well as config reload.
+
+
+### High-Level Design
+Currently hostcfgd controls the services based on the feature table. The feature table has a specific field 'has_timer' for the non essential services which needs to be delayed during the reboot flow. This field will be now replaced by new field called "delay". These services will controlled by hostcfgd.
+During the hostcfgd initialization it will cache these delayed services based on the configuration in the feature table. The hostcfgd will also subscribe to PORT_TABLE in the APPL_DB. Once the switch is initialized and all the ports are created in ASIC and Kernel the PortSyncd will publish PortInitDone key in the APPL_DB. On receiving this key the hostcfgd will go through the delayed services list and enables them.
+
+
+The below diagram explains the sequence when config reload is executed. 
+![](/images/config_reload/Enhance_config_reload.JPG)
+The below diagram explains the new sequence inside hostcfgd.
+![](/images/config_reload/Hostcfgd_flow_for_delayed_services.JPG)
+### SAI API Requirements
+None
+
+### Configuration and management
+
+#### Config command
+
+No new commands are introduced as part of this design.
+
+#### Show command
+
+No new commands are introduced as part of this design
+
+#### DB Migrator
+
+The 'has_timer' field in FEATURE table will be changed to 'delay'. Hence db_migrator is required to modify the configurations to reflect this change.
+
+### YANG model changes
+
+Yang model needs to be updated for FEATURE_TABLE. The 'has_timer' field will be removed and replaced with 'delay'
+
+```
+                leaf delay {

```

Comment:
I suggest:
```suggestion
                leaf delayed {

```
To include an intuitive understanding of a boolean type of this value